src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/user/controllers/UserAuthController.java

```diff
@@ -164,48 +173,56 @@ LoginResponse create(final HttpServletRequest request,
+        token = "";
+
+        PersonEmailVerification personEmailVerification = personEmailVerificationService.create(
+            person, request.getRemoteAddr(), request.getHeader("User-Agent"));
+
         Map<String, Object> params = emailService.getDefaultEmailParameters();
 
         params.put("person", person);
-        params.put("verificationUrl",
-            localInstanceContext.instance().getDomain() + "/api/v3/user/verify_email?token="
-                + "TODO");
+        params.put("verificationUrl", localInstanceContext.instance().getDomain() + "/verify_email/"
+            + personEmailVerification.getToken());
         try {
           final String template_name = EmailTemplatesEnum.VERIFY_EMAIL.toString();
-          emailService.sendEmail(CreateEmailRequest.builder()
-              .to(List.of(person.getEmail()))
+          emailService.saveToQueue(Email.builder()
+              .personRecipients(List.of(person))
               .subject(emailService.getSubjects().get(template_name).getAsString())
-              .body(emailService.formatTextEmailTemplate(template_name,
+              .htmlContent(emailService.formatTextEmailTemplate(template_name,
                   new Context(Locale.getDefault(), params)))
-              .htmlBody(emailService.formatEmailTemplate(template_name,
+              .textContent(emailService.formatEmailTemplate(template_name,
                   new Context(Locale.getDefault(), params)))
               .build());
-          send_verification_email = true;
+
         } catch (Exception e) {
+          personRepository.delete(person);
           throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
               "email_sending_failed");
         }
       }
-    } else {
+    } else if (person.getEmail() != null && !person.getEmail().isEmpty()) {
       Map<String, Object> properties = emailService.getDefaultEmailParameters();
       properties.put("person", person);
 
       try {
         final Context context = new Context(Locale.getDefault(), properties);
 
         final String template_name = EmailTemplatesEnum.REGISTRATION_SUCCESS.toString();
-        emailService.sendEmail(CreateEmailRequest.builder()
-            .to(List.of(person.getEmail()))
+        emailService.saveToQueue(Email.builder()
+            .personRecipients(List.of(person))
             .subject(emailService.getSubjects().get(template_name).getAsString())
-            .body(emailService.formatTextEmailTemplate(template_name, context))
-            .htmlBody(emailService.formatEmailTemplate(template_name, context))
+            .htmlContent(emailService.formatTextEmailTemplate(template_name, context))
+            .textContent(emailService.formatEmailTemplate(template_name, context))
             .build());
-      } catch (Exception e) {
-        // @todo log error?
+      } catch (IOException e) {
+        personRepository.delete(person);
+        logger.error("Error reading Subjects!", e);
```

Comment:
I wonder if we should include more details with this error like account handle

Reply:
I would say no, keep the information only for admins

Reply:
I know the error is for admins; however, I wonder how useful this is as an error message. Just seeing this a lot in the logs makes me wonder if it's one user having the issue or many.

Reply:
This is a critical exception, everyone WILL have it, it only catches IOException and the only thing can throw it is getSubjects() thus if that didnt worked everyone has it

Reply:
Okie dokie